src/main/java/com/zendesk/maxwell/replication/MaxwellReplicator.java

```diff
@@ -214,7 +238,6 @@ private RowMapBuffer getTransactionRows() throws Exception {
 				continue;
 			}
 
-			setReplicatorPosition((AbstractBinlogEventV4) v4Event);
```

Comment:
With this code, the replicator can restart in the middle of a transaction. Since we clear the queue on restart, this is not ideal. I changed it to only set position at the end of transaction, so we can have a clean restart.